include/nbl/video/utilities/IGPUObjectFromAssetConverter.h

```diff
@@ -855,76 +864,117 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUImage** const _begin,
+            toTransferDst.barrier.dstAccessMask = asset::EAF_TRANSFER_WRITE_BIT;
+            toTransferDst.oldLayout = asset::EIL_UNDEFINED;
+            toTransferDst.newLayout = asset::EIL_TRANSFER_DST_OPTIMAL;
+            toTransferDst.srcQueueFamilyIndex = transferFamIx;
+            toTransferDst.dstQueueFamilyIndex = transferFamIx;
+            toTransferDst.image = core::smart_refctd_ptr<video::IGPUImage>(img);
+            toTransferDst.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT; // this probably shoudn't be hardcoded
+            toTransferDst.subresourceRange.baseMipLevel = 0u;
+            toTransferDst.subresourceRange.levelCount = img->getCreationParameters().mipLevels;
+            toTransferDst.subresourceRange.baseArrayLayer = 0u;
+            toTransferDst.subresourceRange.layerCount = cpuimg->getCreationParameters().arrayLayers;
+
+            cmdbuf_transfer->pipelineBarrier(asset::EPSF_TRANSFER_BIT, asset::EPSF_TRANSFER_BIT, asset::EDF_NONE, 0u, nullptr, 1u, &barrier, 1u, &toTransferDst);
+
+            cmdbuf_transfer->copyBufferToImage(buf.get(), img, asset::EIL_TRANSFER_DST_OPTIMAL, cpuimg->getRegions().size(), cpuimg->getRegions().begin());
         }
     };
-    auto cmdComputeMip = [&](const asset::ICPUImage* cpuimg, IGPUImage* img) -> void {
-        if (!needToCompMipsForThisImg(cpuimg))
-            return;
+    auto cmdComputeMip = [&](const asset::ICPUImage* cpuimg, IGPUImage* gpuimg, asset::E_IMAGE_LAYOUT newLayout) -> void
+    {
         // TODO when we have compute shader mips generation:
         /*computeCmdbuf->bindPipeline();
         computeCmdbuf->bindDescriptorSets();
         computeCmdbuf->pushConstants();
         computeCmdbuf->dispatch();*/
+        if (cpuimg->getCreationParameters().mipLevels == gpuimg->getCreationParameters().mipLevels)
+            return;
 
-        uint32_t lowestPresentMip = 1u;
-        for (auto& region : cpuimg->getRegions())
-            lowestPresentMip = (std::max)(lowestPresentMip, region.imageSubresource.mipLevel);
-        // generate temporary image view to make sure we don't screw up any explicit mip levels
-        IGPUImageView::SCreationParams tmpViewParams;
-        tmpViewParams.subresourceRange.levelCount = img->getCreationParameters().mipLevels + 1u - lowestPresentMip;
-        // if not all mip levels have been manually specified
-        if (tmpViewParams.subresourceRange.levelCount > 1u)
+        video::IGPUCommandBuffer::SImageMemoryBarrier barrier = {};
+        barrier.srcQueueFamilyIndex = ~0u;
+        barrier.dstQueueFamilyIndex = ~0u;
+        barrier.image = core::smart_refctd_ptr<video::IGPUImage>(gpuimg);
+        // TODO this is probably wrong (especially in case of depth/stencil formats), but i think i can leave it like this since we'll never have any depth/stencil images loaded (right?)
+        barrier.subresourceRange.aspectMask = cpuimg->getRegions().begin()->imageSubresource.aspectMask; 
```

Comment:
can you even write with a compute shader to a depth or stencil image view directly?

Reply:
but yeah we'll probably not support the automatic mipping of depth or stencil, so this can always be the COLOR aspect.